src/learning/k_means.rs

```diff
@@ -209,14 +209,13 @@ impl<InitAlg: Initializer> KMeansClassifier<InitAlg> {
     /// Used internally within model.
     fn update_centroids(&mut self, inputs: &Matrix<f64>, classes: Vector<usize>) {
         let mut new_centroids = Vec::with_capacity(self.k * inputs.cols());
-        for i in 0..self.k {
-            let vec_i: Vec<usize> = classes.data()
-                .iter()
-                .enumerate()
-                .filter(|&(_, &x)| x == i)
-                .map(|(idx, _)| idx)
-                .collect();
-            
+
+        let mut row_indexes = vec![Vec::new(); self.k];
```

Comment:
By the way, there is [some code](https://github.com/AtheMathmo/rusty-machine/blob/master/src/learning/k_means.rs#L309-L312) I didn't understand.
To me it is missing a `enumerate()`, like the part we're commenting. Else we could rewrite a simpler
```rust 
let mut random_assignments = vec![1; k];
for _ in k..inputs.rows() {
    random_assignments[rng.gen_range(0, k)] += 1;
}
// ...
let vec_i = vec![i; random_assignments[i]];
```